internal/models/sessions.go

```diff
@@ -168,6 +218,35 @@ func FindSessionsByFactorID(tx *storage.Connection, factorID uuid.UUID) ([]*Sess
 	return sessions, nil
 }
 
+// FindAllSessionsForUser finds all of the sessions for a user. If forUpdate is
+// set, it will first lock on the user row which can be used to prevent issues
+// with concurrency. If the lock is acquired, it will return a
+// UserNotFoundError and the operation should be retried. If there are no
+// sessions for the user, a nil result is returned without an error.
+func FindAllSessionsForUser(tx *storage.Connection, userId uuid.UUID, forUpdate bool) ([]*Session, error) {
+	if forUpdate {
```

Comment:
This seems to be a recurring pattern in the codebase (e.g. 
```
if for Update {
   ...
    // acquire corresponding lock
}
```
It's not urgent but we can possibly see if there's sufficient overlap to  factor into a function in the future.

Reply:
I'm afraid it's not very refactorable... most ORMs completely ignore the fact that such statements exist, so you always have to resort to some raw SQL gymnastics. It's also very model dependent, so it's not like it's one SQL statement that saves the day.